.run-controller-gen.sh

```diff
@@ -4,11 +4,25 @@
 # it's the equivalent of `go run sigs.k8s.io/controller-tools/cmd/controller-gen`
 # if you could somehow do that without modifying your go.mod.
 
-current_dir=$(pwd)
-if ! readlink -f . &>/dev/null; then
-    echo "you're probably on OSX.  Please install gnu readlink -- otherwise you're missing the most useful readlink flag."
-    exit 1
-fi
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readlink=$(command -v readlink)
 
-cd $(dirname $(readlink -f ${BASH_SOURCE[0]}))
+check_readlink() {
+    if ! ${readlink} -f &>/dev/null; then
```

Comment:
This will always fail with no file argument.
```console
$ uname -o
GNU/Linux
$ readlink -f
readlink: missing operand
Try 'readlink --help' for more information.
```